ale/rotation.py

```diff
@@ -234,23 +247,47 @@ def _slerp(self, times):
          : 2darray
            The angular velocity vectors
         """
-        vec_times = np.asarray(times)
-        if vec_times.ndim < 1:
-            vec_times = np.asarray([times])
-        elif vec_times.ndim > 1:
+        # Convert non-vector input to vector and check input
+        vec_times = np.atleast_1d(times)
+        if vec_times.ndim > 1:
             raise ValueError('Input times must be either a float or a 1d iterable of floats')
-        if len(self.times) < 2:
-            return Rotation.from_quat(np.repeat(self.quats, len(vec_times), 0)), np.zeros((len(vec_times), 3))
+
+        # Compute constant angular velocity for interpolation intervals
+        avs = np.zeros((len(self.times) + 1, 3))
+        if len(self.times) > 1:
+            steps = self.times[1:] - self.times[:-1]
+            rotvecs = (self._rots[1:] * self._rots[:-1].inv()).as_rotvec()
+            avs[1:-1] = rotvecs / steps[:, None]
+
+        # If available use actual angular velocity for extrapolation
+        # Otherwise use the adjacent interpolation interval
+        if self.av is not None:
+            avs[0] = self.av[0]
+            avs[-1] = self.av[-1]
```

Comment:
Does the angular velocity array that is stored on the rotation class only have two values? I guess my bigger question is why are we interpolating the angular velocity between 0 and N if we have them stored on the object?

Reply:
The angular velocity at each time is stored in the rotation but we don't use them for rotation interpolation. I did testing and it turns out using the stored angular velocities actually results in worse rotation interpolation. The angular velocity is not constant between rotations, so using it for SLERP results in discontinuities in the interpolated rotations. Using a more complex interpolation scheme is more work than we actually need.

This is why I'm not 100% happy with this, but it appears to match what both NAIF and ISIS actually use.

Reply:
These two values are set because they are the angular velocities used to extrapolate rotations. Since we cannot compute an angular velocity here we either use the stored one or the computed values from the first/last interval.